services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala

```diff
@@ -17,83 +18,161 @@ class RunServiceTest extends AsyncWordSpec with Matchers with MockitoSugar {
+    )
+
     "addRun" should {
 
       "returns run id" taggedAs Service in {
-        val runId = TestRunUtils.getDummyRunId
-        val project = TestProjectUtils.getDummyProject()
-        val projectId = project.projectId
-        val userId = TestUserUtils.getDummyUserId
-        val run = TestRunUtils.getDummyRun(runId = runId, projectId = projectId, userId = userId)
-        val runCreateRequest = RunCreateRequest(
-          projectId = projectId,
-          projectVersion = run.projectVersion,
-          results = run.results
-        )
-
+        val runCreateRequest = RunCreateRequest(projectVersion = run.projectVersion, results = run.results)
         when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
         when(runRepository.addRun(any[Run])).thenReturn(Future.successful(runId))
-        runService.addRun(runCreateRequest, userId).map {
+
+        runService.addRun(runCreateRequest, projectId, userId).map {
           _ shouldBe runId
         }
       }
     }
 
-    "getRunById" should {
-
-      "returns run entity if the entity exists" taggedAs Service in {
+    "getRunById when user is the owner" should {
 
-        val runId = TestRunUtils.getDummyRunId
-        val run = TestRunUtils.getDummyRun(runId)
+      "return run if run exists" taggedAs Service in {
+        when(projectService.getUserProjectById(projectId, run.userId)).thenReturn(Future.successful(project))
         when(runRepository.getRunByIdAndUser(runId, run.userId)).thenReturn(Future.successful(Some(run)))
 
-        runService.getRunByIdAndUser(runId, run.userId).map { _ shouldBe Some(run) }
+        runService.getRunByIdAndUser(runId, projectId, run.userId).map { _ shouldBe Some(run) }
+      }
+
+      "return none if the run is not found" taggedAs Service in {
+        when(projectService.getUserProjectById(projectId, run.userId)).thenReturn(Future.successful(project))
+        when(runRepository.getRunByIdAndUser(runId, run.userId)).thenReturn(Future(None))
+
+        runService.getRunByIdAndUser(runId, projectId, run.userId).map { _ shouldBe None }
+      }
+
+      "return none if run belongs to another project of the same user" taggedAs Service in {
+        val anotherProjectRun = TestRunUtils.getDummyRun(userId = run.userId)
+        when(projectService.getUserProjectById(projectId, run.userId)).thenReturn(Future.successful(project))
+        when(runRepository.getRunByIdAndUser(anotherProjectRun.runId, run.userId))
+          .thenReturn(Future.successful(Some(anotherProjectRun)))
+
+        runService.getRunByIdAndUser(anotherProjectRun.runId, projectId, run.userId).map {
+          _ shouldBe None
+        }
+      }
+    }
+
+    "getRunById when user is NOT the owner" should {
+
+      "fail with access denied exception" taggedAs Service in {
+        val error = ProjectAccessDeniedException()
+        when(runRepository.getRunByIdAndUser(runId, userId)).thenReturn(Future.failed(error))
```

Comment:
This test makes no sense. Our repository is not supposed to return `ProjectAccessDeniedException`

Reply:
Line 89 deleted